### PR TITLE
Consumer timeout wait

### PR DESCRIPTION
**Pull request by kirkbaly**
Added the ability to set a timeout_wait flag for consumers. This ensures the consumer appropriately terminates when its connection is lost, rather than waiting for an idle timeout or a max execution timeout.

### CODE REVIEW

File: RabbitMq/Consumer.php [hunk 1]
```diff
@@ -76,29 +84,35 @@ public function consume($msgAmount)
...
-                    $this->dispatchEvent(OnIdleEvent::NAME, $idleEvent);
+                    $now = time();
 
-                    if ($idleEvent->isForceStop()) {
-                        if (null !== $this->getIdleTimeoutExitCode()) {
-                            return $this->getIdleTimeoutExitCode();
-                        } else {
-                            throw $e;
+                    if ($this->gracefulMaxExecutionDateTime
+                        && $this->gracefulMaxExecutionDateTime <= new \DateTime("@$now")
```
Comment (ramunasd) [inline RabbitMq/Consumer.php:101]: Why not just use timestamps everywhere? I mean there is no point to use \DateTime here.

**Comment by ramunasd** [comment]
I see there is too much conflicts since last update. Please rebase and reopen as new PR.

File: RabbitMq/Consumer.php (hunk 1 above)
Comment (kirkbaly) [reply RabbitMq/Consumer.php:101]: Yes, agreed that timestamps would be preferred. I introduced $lastActivityDateTime as \DateTime to match $gracefulMaxExecutionDateTime which was already in place.

**Comment by kirkbaly** [comment]
This PR was opened about 3 years ago back when the repo was not being maintained, so it is perhaps just a little out of date :)

However, it does appear that my changes were introduced by @constantable in 5542098e2cac53cef24dae1ded3772202c5d3411. It looks like they are already in master, so there is no need for this PR or a new one.

Thanks!

**Pull request closed without merging**